### PR TITLE
Fix TC-OCC-3.1 and TC-OCC-3.2 runtime errors

### LINKED ISSUE #34730: [CERT-TEST-FAILURE] [TC-OCC-3.1, 3.2] - Python Script Failure
### Feature Area

Other

### Test Case

TC-OCC-3.1
TC-OCC-3.2

### Reproduction steps

1. Command used to launch the DUT : `./chip-all-clusters-app `
2. Command used to run the python script : `python3 TC_OCC_3_2.py --commissioning-method on-network --discriminator 3840 --passcode 20202021 --storage-path admin_storage.json  --paa-trust-store-path ../../credentials/development/paa-root-certs/`

While executing the TC-OCC-3.2 python script, Test case is failing due to the `AttributeError: module 'chip.ChipDeviceCtrl' has no attribute 'ReadAttribute'` error. . 

In Python script line number 142 `ChipDeviceCtrl` has to be updated to Actual controller object.

> [MatterTest] 08-02 08:10:37.058 INFO 		***** Test Step 2 : TH establishes a wildcard subscription to all attributes on Occupancy Sensing Cluster on the endpoint under test. Subscription min interval = 0 and max interval = 30 seconds.
> [MatterTest] 08-02 08:10:37.061 ERROR Exception occurred in test_TC_OCC_3_2.
> Traceback (most recent call last):
>   File "/home/ubuntu/master_CNTRL/connectedhomeip/no/lib/python3.10/site-packages/mobly/base_test.py", line 783, in exec_one_test
>     test_method()
>   File "/home/ubuntu/master_CNTRL/connectedhomeip/src/python_testing/matter_testing_support.py", line 1671, in async_runner
>     return _async_runner(body, self, *args, **kwargs)
>   File "/home/ubuntu/master_CNTRL/connectedhomeip/src/python_testing/matter_testing_support.py", line 1659, in _async_runner
>     return asyncio.run(runner_with_timeout)
>   File "/usr/lib/python3.10/asyncio/runners.py", line 44, in run
>     return loop.run_until_complete(main)
>   File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
>     return future.result()
>   File "/usr/lib/python3.10/asyncio/tasks.py", line 445, in wait_for
>     return fut.result()
>   File "/home/ubuntu/master_CNTRL/connectedhomeip/src/python_testing/TC_OCC_3_2.py", line 142, in test_TC_OCC_3_2
>     await attrib_listener.start(ChipDeviceCtrl, node_id, endpoint=endpoint_id)
>   File "/home/ubuntu/master_CNTRL/connectedhomeip/src/python_testing/matter_testing_support.py", line 360, in start
>     self._subscription = await dev_ctrl.ReadAttribute(
> AttributeError: module 'chip.ChipDeviceCtrl' has no attribute 'ReadAttribute'
> [MatterTest] 08-02 08:10:37.068 INFO 		***** Test Failure : 
> 

### Anything else?

**Python Script References:**
- https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/TC_OCC_3_2.py

image

**Reference Log:**
[TC-OCC-3.2.txt](https://github.com/user-attachments/files/16468517/TC-OCC-3.2.txt)
[TC-OCC-3.1.txt](https://github.com/user-attachments/files/16468522/TC-OCC-3.1.txt)

### PR DESCRIPTION
**Pull request by tcarmelveilleux**
- Refactor common code to be clearer in TC-OCC-3.1/3.2
- Fix runtime errors
- Move `await_sequence_of_reports` to matter_testing_support.py
- Make a helper to write attributes: `write_single_attribute`

Fixes #34730

Testing done:

- TC-OCC-3.1 and 3.2 no longer break locally (CI integration pending named pipe update in follow-up)
- TC_SWTCH still works

### CODE REVIEW

File: src/python_testing/TC_OCC_3_1.py [hunk 1]
```diff
@@ -29,19 +29,29 @@
...
 import chip.clusters as Clusters
-from chip import ChipDeviceCtrl
 from chip.interaction_model import Status
 from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
 from mobly import asserts
 
 
 class TC_OCC_3_1(MatterBaseTest):
-    async def read_occ_attribute_expect_success(self, endpoint, attribute):
+    async def read_occ_attribute_expect_success(self, attribute):
```
Comment (cecille) [inline src/python_testing/TC_OCC_3_1.py:41]: Not a problem to expose explicitly, but read_single_attribute_check_success defaults to the matter_test_config.endpoint

**Review by jaehs6sam — COMMENTED** [review]
All revised files look good to me.

File: src/python_testing/matter_testing_support.py [hunk 2]
```diff
@@ -331,6 +332,58 @@ def wait_for_report(self):
             asserts.fail("[AttributeChangeCallback] Attribute {expected_attribute} not found in returned report")
 
 
+def await_sequence_of_reports(report_queue: queue.Queue, endpoint_id: int, attribute: TypedAttributePath, sequence: list[Any], timeout_sec: float):
```
Comment (cecille) [inline src/python_testing/matter_testing_support.py:335]: Q - why not make this a function of the AttributeChangeAccumulator?

**Review by cecille — APPROVED** [review]

File: src/python_testing/matter_testing_support.py (hunk 2 above)
Comment (tcarmelveilleux) [reply src/python_testing/matter_testing_support.py:335]: We could add a method that calls this. Will do as a follow-up.

File: src/python_testing/TC_OCC_3_1.py (hunk 1 above)
Comment (tcarmelveilleux) [reply src/python_testing/TC_OCC_3_1.py:41]: Will do as follow-up.

**Review by woody-apple — APPROVED** [review]

**Pull request merged**